[COST-5805] Update docs with what's new in v4.0.0 (#581)
project-koku/koku-metrics-operator: docs/report-fields-description.md

* To enable the collection ROS (Resource Optimization) metrics, ensure that the namespace(s) are labeled with `insights_cost_management_optimizations='true'`.

    * Queries responsible for collecting ROS metrics are identified in the `QueryMap` with the prefix `ros:` and include a specific filter to target the appropriately labeled namespaces:
        ```
        kube_namespace_labels{label_insights_cost_management_optimizations='true', namespace!~'kube-.*|openshift|openshift-.*'}
        ```

### 1. Common Fields

* `report_period_start`: The start timestamp of the reporting period.
* `report_period_end`: The end timestamp of the reporting period.
* `interval_start`: The start timestamp of the reporting interval.
* `interval_end`: The end timestamp of the reporting interval.

These common fields are included in all the reports and provide temporal information about the reporting period and interval.

---

### 2. Container Metrics

Fields for metrics related to containers:

* `container_name`: The name of the container.
* `pod`: The name of the pod that contains the container.

> * `owner_name`: The name of the owner entity (e.g., Deployment, StatefulSet) associated with the container.

Reviewer: ```suggestion
* `owner_name`: The name of the owner entity that is associated with the container. For example, Deployment or StatefulSet.
```
note: should deployemnt and StatefulSet be in monospace?

Fields for metrics related to pods:

* `node`: The name of the node.
* `namespace`: The namespace of the pod.
* `pod`: The name of the pod.
* `pod_usage_cpu_core_seconds`: CPU core seconds used by the pod.
* `pod_request_cpu_core_seconds`: CPU core seconds requested by the pod.
* `pod_limit_cpu_core_seconds`: CPU core seconds limited for the pod.
* `pod_usage_memory_byte_seconds`: Memory byte seconds used by the pod.
* `pod_request_memory_byte_seconds`: Memory byte seconds requested by the pod.
* `pod_limit_memory_byte_seconds`: Memory byte seconds limited for the pod.
* `node_capacity_cpu_cores`: CPU cores capacity of the node.
* `node_capacity_cpu_core_seconds`: CPU core seconds capacity of the node.
* `node_capacity_memory_bytes`: Memory bytes capacity of the node.
* `node_capacity_memory_byte_seconds`: Memory byte seconds capacity of the node.
* `node_role`: The role of the node.
* `resource_id`: The ID of the resource.
* `pod_labels`: Labels associated with the pod.

---

### 6. Namespace Metrics Report

Fields for metrics related to namespaces:

* `namespace`: The namespace.
* `namespace_labels`: Labels associated with the namespace.

---

### 7. VM Metrics

> Fields for metrics related to running virtual machines and their associated resources:

Reviewer: ```suggestion
Fields for metrics related to running virtual machines (VMs) and their associated resources:
```

> * `node`: The name of the node where the VM instance (VMI) is currently running.

Reviewer: ```suggestion
* `node`: The name of the node where the virtual machine instance (VMI) is currently running.
```

* `namespace`: The namespace where the VM (VM) is defined.
* `resource_id`: The unique identifier of the VM on the node. This is derived from the `provider_id` of the node, specifically the segment after the last `/` character (e.g., if `provider_id` is `aws:///us-east-1a/i-0abcdef1234567890`, `resource_id` would be `i-0abcdef1234567890`).
* `vm_name`: The name of the VM.
* `vm_instance_type`: The instance type associated with the VM, if defined.
* `vm_os`: The operating system reported by the VM's info.

> * `vm_guest_os_arch`: The guest operating system architecture reported by the VM (e.g., `x86_64`).

Reviewer: similar suggestions as above for e.g. --> For example,

* `vm_guest_os_name`: The guest operating system name reported by the VM (e.g., `RHEL`, `Fedora`).
* `vm_guest_os_version`: The guest operating system version ID reported by the VM (e.g., `8.6`).
* `vm_uptime_total_seconds`: The total uptime of the VMI in seconds since it started.

> * `vm_cpu_limit_cores`: CPU core limit configured, representing the maximum number of cores the VM can use.

Reviewer: Suggestion to make more of a sentence like "The CPU core limit that is configured" or something like that
